module/communication/mqtt/src/test/java/org/openbase/jul/communication/mqtt/IntegrationTest.kt

```diff
@@ -109,4 +126,42 @@ class IntegrationTest : AbstractIntegrationTest() {
             lock.wait()
         }
     }
+
+    @Test
+    fun `test parallel method call`() {
+        val instance = MethodMock()
+        val scope = ScopeProcessor.concat(scope, ScopeProcessor.generateScope("parallel_execution"))
+
+        val rpcServer = RPCServerImpl(scope, config)
+        rpcServer.registerMethod(MethodMock::blockingRPC, instance)
+        rpcServer.registerMethod(MethodMock::ping, instance)
+        rpcServer.activate()
+        rpcServer.getActivationFuture()!!.get()
+
+        val rpcClient = RPCClientImpl(scope, config)
+
+        instance.lock.lock()
+        try {
+            val expectedMsg = "hello"
+            val blockingFuture = rpcClient.callMethod(instance::blockingRPC.name, String::class, expectedMsg)
+
+            val time = 42L
+            try {
+                val pingResult =
```

Comment:
try to avoid var creation since its often interrupting the "kotlin sentence semantics"